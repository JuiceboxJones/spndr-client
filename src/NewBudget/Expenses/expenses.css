#header h2 {
  margin-top: 0px;
}

.expensesContainer{
  padding: 15px;
}

.current_monthly_expenses{
  width: 250px;
  margin: 0 auto;
}

.form_container{
  margin: 0 auto;
  width: 33%;
  margin-top: 5%;
  border-radius: 8px;
  background: #f5f5f5;
  padding-top: 40px;
  height: 450px;
}

.expense_list{
  position: relative;
  margin: 30px;
}

.button_container{
  position: fixed;
  width: 450px;
  bottom: 220px;
}

#back_button{
  border: none;
  background:#A7C4C0;
  border-radius: 5px 0px 0px 5px;
  font-weight: bold;
  width: 100px;
  height: 25px;
  margin: 5px;
  cursor: pointer;
}

#next_button{
  border: none;
  background:#A7C4C0;
  border-radius: 0px 5px 5px 0px;
  font-weight: bold;
  width: 100px;
  height: 25px;
  margin: 5px;
  cursor: pointer;
}

#next_button:hover{
  background: #024e2e9c;
  color:white;
} 
#back_button:hover{
  background: #024e2e9c;
  color:white;
}
#delete{
  float: right;
  margin-right: 20px;
  margin-top: 2px;
  border: none;
  background: none;
  cursor: pointer;
}
#delete:hover{
  font-size: 95%;
  text-shadow: 7px -3px 3px rgba(0, 0, 0, 0.644);
}

#add_expense{
  margin: 15px;
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background:#A7C4C0;
  cursor: pointer;
  font-weight: bold;
}
#add_expense:hover{
  background: #024e2e9c;
  color:white;
}

.expense_list li{
  border-radius: 2px;
  background: #dbdada;
  font-size: 17px;
  height: 25px;
  font-weight: bold;
  color: rgb(209, 0, 0);
  padding: 1px;
  margin: 5px;
}
span{
  margin-left: 10px;
}
