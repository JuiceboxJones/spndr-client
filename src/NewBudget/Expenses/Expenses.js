import React, { Component } from 'react';
import ApiService from '../../services/api-fetch-services';
import HistoryHelper from '../../History/History';
import './expenses.css'

class Expenses extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }
  
  state = {
    expenses: []
  };

componentDidMount(){
  ApiService.getExpenses()
  .then(data => {
    this.setState({expenses: data})
  })
}

  handleAddExpense = e => {
    e.preventDefault();
    const addExpense = {
      name: e.target.name.value,
      amount: e.target.amount.value };
    ApiService.postExpenses(addExpense).then(exp => 
    {this.setState({ expenses: [...this.state.expenses, exp] });})
    e.target.reset();
    e.target.name.focus()
  };

  handleDelete(e) {
    ApiService.deleteExpense(e)
    this.handleDeleteitem(e)
  }
  
  handleDeleteitem = (e) => {
    const expenses = this.state.expenses
    .filter(item => item.id !== e)
    this.setState({expenses})
  }

  handleNext(){
    HistoryHelper.historyGoTo('/wishlist', this.props)
  }
  handleBack(){
    HistoryHelper.historyGoBack(this.props)
  }

  render() {

    const handleList = this.state.expenses
    .map((exp, index) => {
      return <li key={exp.id || index }>
      {exp.name} <span/> ${exp.amount} <button type='button' id='delete' onClick={() => this.handleDelete(exp.id)}>
      <span id='delete' role="img" aria-label="delete">‚ùå</span></button>
      </li>
    })

    return (
      <div className='expensesContainer'>
      <header id='header'>
        <h2>Add Expenses</h2>
        <p>Step 2 of 3</p>
      </header>
      <div className='form_container'>
        <form className="current_expenses" onSubmit={e => this.handleAddExpense(e)}>
        <div className="current_monthly_expenses">
          <label htmlFor="current_name">Expense name</label>
          <input type="text" required name="name" id="current_name" />
          <label htmlFor="current_amount">Amount</label>
          <input type="number" step='0.01' required name="amount" id="current_amount" />
          <button id='add_expense' type="submit">Add Expense</button>
        </div>
        <div className='expense_list'>
        <ul>{handleList}</ul>
        </div>
        <div className='button_container'>
        <button type='button' id='back_button' onClick={(e) => this.handleBack(e)}>Back</button>
        <button type="button" id='next_button' onClick={(e) => this.handleNext(e)}>Next</button>
        </div>
      </form>
      </div>
      </div>

    );
  }
}

export default Expenses;
