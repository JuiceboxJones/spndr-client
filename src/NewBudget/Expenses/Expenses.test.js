import React from 'react';
import ReactDOM from 'react-dom';
import Expenses from './Expenses'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Expenses />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Expenses />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});