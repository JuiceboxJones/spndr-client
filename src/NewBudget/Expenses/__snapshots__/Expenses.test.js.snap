// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="expensesContainer"
>
  <header
    id="header"
  >
    <h2>
      Add Expenses
    </h2>
    <p>
      Step 2 of 3
    </p>
  </header>
  <div
    className="form_container"
  >
    <form
      className="current_expenses"
      onSubmit={[Function]}
    >
      <div
        className="current_monthly_expenses"
      >
        <label
          htmlFor="current_expense_name"
        >
          Expense name
        </label>
        <input
          id="current_name"
          name="name"
          required={true}
          type="text"
        />
        <label
          htmlFor="current_expense_amount"
        >
          Amount
        </label>
        <input
          id="current_amount"
          name="amount"
          required={true}
          step="0.01"
          type="number"
        />
        <button
          id="add_expense"
          type="submit"
        >
          Add Expense
        </button>
      </div>
      <div
        className="expense_list"
      >
        <ul />
      </div>
      <div
        className="button_container"
      >
        <button
          id="back_button"
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
        <button
          id="next_button"
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
    </form>
  </div>
</div>
`;
